changelog/new_style_hash_conversion_cop.md

```diff
@@ -0,0 +1 @@
+* [#9478](https://github.com/rubocop-hq/rubocop/pull/9478): Add new `Style/LegacyHash` cop. ([@zverok][])
```

Comment:
Just noticed that we need to update the name here as well.

Reply:
@bbatsov fixed